### PR TITLE
Lazy loading for routes and webpack optimizations

### PR DESCRIPTION
**Pull request by jesusbotella**
While configuring a new route for Direct DB connection page (not in this branch), I saw that there were so many routes within the same file. So, I split them into several files and leveraged the opportunity to add lazy loading to all of the page components.

Kinda related to https://github.com/CartoDB/cartodb/pull/14606

### CODE REVIEW

File: lib/assets/javascripts/new-dashboard/router/routes/connected_apps.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Lazy Pages
+const ConnectedApps = () => import('new-dashboard/pages/Apps/ConnectedApps');
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/router/routes/connected_apps.js]: Really nice idiom!

Can we get this working without adding any plugin to our webpack? 
I read about https://babeljs.io/docs/en/babel-plugin-syntax-dynamic-import/ but I'm not sure...

File: lib/assets/javascripts/new-dashboard/router/routes/home.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Lazy Pages
+import Home from 'new-dashboard/pages/Home/Home';
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/router/routes/home.js]: Should this be?
```
const Home = () => import('new-dashboard/pages/Home/Home');
```

File: lib/assets/javascripts/new-dashboard/router/routes/connected_apps.js (hunk 1 above)
Comment (jesusbotella) [reply lib/assets/javascripts/new-dashboard/router/routes/connected_apps.js]: It does work right now without adding the plugin, but I don't know if it works with IE11. I guess not, so I think I am gonna test it out add that plugin if it doesn't work.

File: lib/assets/javascripts/new-dashboard/router/routes/home.js (hunk 2 above)
Comment (jesusbotella) [reply lib/assets/javascripts/new-dashboard/router/routes/home.js]: I thought that the home is going to be visited anyway so I didn't lazy load it. What do you think?

Comment (VictorVelarde) [reply lib/assets/javascripts/new-dashboard/router/routes/home.js]: Sounds legit, then let's replace the comment to specify it is not lazy

File: webpack/v4/webpack.prod.config.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ module.exports = merge(baseConfig, {
     })
   ],
   optimization: {
+    nodeEnv: 'production',
```
Comment (VictorVelarde) [inline webpack/v4/webpack.prod.config.js]: Do we need to set nodeEnv when we already have `mode: 'production'` in the file?

File: lib/assets/javascripts/new-dashboard/.babelrc [hunk 4]
```diff
@@ -1,19 +1,42 @@
 {
   "presets": [
-    ["env", {
+    ["@babel/env", {
       "modules": false,
       "targets": {
         "browsers": ["> 1%", "last 2 versions", "not ie < 11"]
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/.babelrc]: I'm not sure if browserlist is compatible with ALL the tooling we have now, so we could have just a .browserslistrc

File: package.json [hunk 5]
```diff
@@ -74,21 +77,19 @@
...
-    "babel-plugin-syntax-jsx": "^6.18.0",
-    "babel-plugin-transform-es2015-modules-commonjs": "^6.26.2",
-    "babel-plugin-transform-object-assign": "^6.22.0",
-    "babel-plugin-transform-object-rest-spread": "^6.26.0",
-    "babel-plugin-transform-runtime": "^6.23.0",
-    "babel-plugin-transform-vue-jsx": "^3.7.0",
-    "babel-preset-env": "^1.7.0",
-    "babel-preset-es2015": "6.24.1",
-    "babel-preset-stage-2": "^6.24.1",
+    "@babel/plugin-syntax-jsx": "^7.8.3",
```
Comment (VictorVelarde) [inline package.json]: ✨  Nice boy scout!

File: babel.config.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    "@babel/plugin-transform-classes",
+    "@babel/plugin-transform-object-assign",
+    [
+      "@babel/plugin-transform-runtime",
+      {
+        "corejs": 3
+      }
+    ]
+  ],
+  "sourceType": "unambiguous",
```
Comment (VictorVelarde) [inline babel.config.json:22]: Any particular problem found before this `sourceType` setting or it is just "recommended"?

File: webpack/v4/webpack.base.config.js [hunk 7]
```diff
@@ -28,6 +29,8 @@ module.exports = {
   },
   devtool: 'source-map',
   plugins: [
+    new webpack.ContextReplacementPlugin(/moment[/\\]locale$/, /en/),
```
Comment (VictorVelarde) [inline webpack/v4/webpack.base.config.js:32]: Nice addition, I guess we're saving some bytes there...

File: babel.config.json (hunk 6 above)
Comment (jesusbotella) [reply babel.config.json:22]: We need to set it to `unambiguous` because we mix different module formats.

https://babeljs.io/docs/en/options#sourcetype

**Comment by VictorVelarde** [comment]
@jesusbotella Really nice job here. Thx for the effort on updates and improvements

**Pull request merged**